ImageHandler.py

```diff
@@ -0,0 +1,113 @@
+            image_model_id = "timbrooks/instruct-pix2pix"
+            self.image_pipe = StableDiffusionInstructPix2PixPipeline.from_pretrained(
+                image_model_id, torch_dtype=torch.float16, safety_checker=None
+            ).to(self.device)
+            self.image_pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(self.image_pipe.scheduler.config)
+
+            # Text to Image model
+            # https://huggingface.co/runwayml/stable-diffusion-v1-5
+            # https://github.com/runwayml/stable-diffusion
+            text_model_id = "runwayml/stable-diffusion-v1-5"
+            self.text_pipe = AutoPipelineForText2Image.from_pretrained(
+                text_model_id, torch_dtype=torch.float16, safety_checker=None
+            ).to(self.device)
+            self.text_pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(self.image_pipe.scheduler.config)
+            
+            # Initialize the asynchronous queue and thread
+            self.queue = queue.Queue()
+            self.thread = threading.Thread(target=self._process_queue)
+            self.thread.daemon = True
+            self.thread.start()
+
+    # Function to get image from url
+    def get_image_from_url(self, url: str) -> PIL.Image.Image:
+        image = PIL.Image.open(requests.get(url, stream=True).raw)
+        image = PIL.ImageOps.exif_transpose(image)
+        image = image.convert("RGB")
+        return image
+    
+    # Function to add image to image task to queue
+    def enqueue_image_to_image(self, init_image: Union[PIL.Image.Image, str], prompt = 'make photograph', num_inference_steps=None, guidance_scale=None):
+        if isinstance(init_image, str):
+            init_image = self.get_image_from_url(init_image)
+
+        kwargs = {'image': init_image, 'prompt': prompt}
+
+        if num_inference_steps is not None:
+            kwargs['num_inference_steps'] = num_inference_steps
+        if guidance_scale is not None:
+            kwargs['image_guidance_scale'] = guidance_scale
+
+        task = {'type': 'image', 'kwargs': kwargs}
+        self.queue.put(task)
+        print(f"\nEnqueued: {prompt}")
+
+    # Function to add text to image task to queue
+    def enqueue_prompt_to_image(self, prompt: str, negative_prompt = None, num_inference_steps=None, guidance_scale=None) -> PIL.Image.Image:
+        kwargs = {'prompt': prompt}
+        if negative_prompt is not None:
+            kwargs['negative_prompt'] = negative_prompt
+        if num_inference_steps is not None:
+            kwargs['num_inference_steps'] = num_inference_steps
+        if guidance_scale is not None:
+            kwargs['guidance_scale'] = guidance_scale
+
+        task = {'type': 'prompt', 'kwargs': kwargs}
+        self.queue.put(task)
+        print(f"\nEnqueued: {prompt}")
+    
+    # Continuously process the queue
+    def _process_queue(self):
+        while True:
+            task = self.queue.get()
+            if task is None:
```

Comment:
nit: could it be best practice for readability to put the positive case in the conditional?
```suggestion
            if task:
                print(f"\nDequeued: {task['kwargs']['prompt']}")
                self.process(task)
                self.queue.task_done()
            else:
                 break
```